### PR TITLE
Adding ServiceInfo documentation

### PR DESCRIPTION
**Pull request by DukeDavis12**
Adding the updated ServiceInfo documentation to the reference guide.

Signed-off-by: Davis Benny <davis.benny@intel.com>

### CODE REVIEW

File: docs/iot-platform-sdk/running-the-demo.md [hunk 1]
```diff
@@ -454,47 +454,111 @@ The above directory contains several subdirectories, each representing a device,
...
     "msg": "exec",
     "valueLen": -1,
-    "valueId": "sh-linux64",
+    "valueId": "binsh-linux64",
     "enc": "base64"
   }
 ]
 ```
+
+!!! Note
```
Comment (Darshini-Parikh93) [inline docs/iot-platform-sdk/running-the-demo.md:467]: I would suggest to follow the following syntax for adding Note: **NOTE** in place of !!!Note.
This is the syntax followed across several README files.

File: docs/iot-platform-sdk/running-the-demo.md [hunk 2]
```diff
@@ -454,47 +454,111 @@ The above directory contains several subdirectories, each representing a device,
...
+     The ordering of message blocks are important for **Client-SDK** / **Client-CSDK TPM** device. The “filedesc” block must always be followed by “write” block and if any executable script is transfered then the “exec” block must be immediately followed after both “filedesc” and “write” blocks.
```
Comment (Darshini-Parikh93) [inline docs/iot-platform-sdk/running-the-demo.md]: For words quoted in double quotes: enclose them in `word`
eg: instead of "filedesc" make it `filedesc`

File: docs/iot-platform-sdk/running-the-demo.md [hunk 3]
```diff
@@ -508,9 +572,9 @@ Given an OwnershipVoucher file with a unique deviceId/GUID, the following needs
 
 The following is a sample Python\* script that, given an Owner voucher file as an argument, extracts the GUID and creates the previously mentioned device directory structure.
 
-This sample script operates under the assumption that both the script and voucher files are placed at \<sdo-iot-platform-sdk-root>/demo/ocs/config/db/v1/devices, and that this is the current working directory:
+This sample script operates under the assumption that both the script and voucher files are placed at <sdo-iot-platform-sdk-root>/demo/ocs/config/db/v1/devices, and that this is the current working directory:
```
Comment (Darshini-Parikh93) [inline docs/iot-platform-sdk/running-the-demo.md]: enclose the files location in `file_location` as is done on line 462 in the PR.
Please apply this to all the file locations referenced in this PR

File: docs/iot-platform-sdk/running-the-demo.md (hunk 1 above)
Comment (DukeDavis12) [reply docs/iot-platform-sdk/running-the-demo.md:467]: !!! NOTE  is used to render an alert box by MKDocs. So I think it better to stick with the existing **!!! NOTE**

File: docs/iot-platform-sdk/running-the-demo.md (hunk 2 above)
Comment (DukeDavis12) [reply docs/iot-platform-sdk/running-the-demo.md]: I will enclose it in a word.

File: docs/iot-platform-sdk/running-the-demo.md (hunk 3 above)
Comment (DukeDavis12) [reply docs/iot-platform-sdk/running-the-demo.md]: I will enclose the paths with file_location attribute.

File: docs/iot-platform-sdk/running-the-demo.md (hunk 1 above)
Comment (Darshini-Parikh93) [reply docs/iot-platform-sdk/running-the-demo.md:467]: ok. I suggested because the alert box is not rendering in git.

File: docs/iot-platform-sdk/running-the-demo.md (hunk 3 above)
Comment (KiranSukhavasi) [inline docs/iot-platform-sdk/running-the-demo.md]: location to place the script is not correct. please rectify this path.

File: docs/iot-platform-sdk/running-the-demo.md [hunk 4]
```diff
@@ -454,47 +454,111 @@ The above directory contains several subdirectories, each representing a device,
...
+      "msg": "exec",
+      "valueLen": -1,
+      "valueId": "binsh-script2",
+      "enc": "base64"
+    }
+]
+```
+!!! Note
+     Using a single “exec” block you cannot execute multiple scripts on the Device.
+     For devices except Client-SDK and Client-CSDK TPM, the above order is not mandatory, we can first define “filedesc” and “write” blocks for both scripts and then define “exec” blocks later.
```
Comment (prateekcr) [inline docs/iot-platform-sdk/running-the-demo.md]: Combine this Note with the Note present in Line 507.

File: docs/iot-platform-sdk/running-the-demo.md [hunk 5]
```diff
@@ -454,47 +454,111 @@ The above directory contains several subdirectories, each representing a device,
...
+    "valueId": "binsh-linux64",
     "enc": "base64"
   }
 ]
 ```
+
+!!! Note
+     The ordering of message blocks are important for **Client-SDK** / **Client-CSDK TPM** device. The “filedesc” block must always be followed by “write” block and if any executable script is transfered then the “exec” block must be immediately followed after both “filedesc” and “write” blocks.
+
+For the scenario of transferring multiple executables to the Device, then the following order of blocks should be adhered.
```
Comment (prateekcr) [inline docs/iot-platform-sdk/running-the-demo.md]: Suggest rewording the statement a bit:
For the scenario of transferring multiple executables to the Device, following order of blocks should be maintained.

File: docs/iot-platform-sdk/running-the-demo.md (hunk 4 above)
Comment (DukeDavis12) [reply docs/iot-platform-sdk/running-the-demo.md]: Done !

File: docs/iot-platform-sdk/running-the-demo.md [hunk 6]
```diff
@@ -454,63 +454,124 @@ The above directory contains several subdirectories, each representing a device,
...
     "valueLen": -1,
-    "valueId": "sh-linux64",
+    "valueId": "binsh-linux64",
     "enc": "base64"
   }
 ]
 ```
+
+!!! Note
+     The ordering of message blocks are important for **Client-SDK** / **Client-CSDK TPM** device. The `filedesc` block must always be followed by `write` block and if any executable script is transfered then the `exec` block must be immediately followed after both `filedesc` and `write` blocks. For devices except Client-SDK and Client-CSDK TPM, the above order is not mandatory, we can first define `filedesc` and `write` blocks for both scripts and then define `exec` blocks later.
```
Comment (prateekcr) [inline docs/iot-platform-sdk/running-the-demo.md]: Replace 'Client-CSDK TPM' with 'Client-SDK TPM'

Comment (DukeDavis12) [reply docs/iot-platform-sdk/running-the-demo.md]: Done

**Review by prateekcr — DISMISSED** [review]

File: docs/iot-platform-sdk/running-the-demo.md (hunk 5 above)
Comment (DukeDavis12) [reply docs/iot-platform-sdk/running-the-demo.md]: Done

File: docs/iot-platform-sdk/running-the-demo.md (hunk 3 above)
Comment (DukeDavis12) [reply docs/iot-platform-sdk/running-the-demo.md]: Path points to the voucher extension python script.

**Review by Darshini-Parikh93 — DISMISSED** [review]

**Review by KiranSukhavasi — APPROVED** [review]

File: docs/iot-platform-sdk/running-the-demo.md [hunk 7]
```diff
@@ -454,63 +454,124 @@ The above directory contains several subdirectories, each representing a device,
 
 The demo onboarding process demonstrates download of a script file and a binary file. Once the files are downloaded, the script validates the checksum of the binary file and creates result.txt to store the result.
 
-These files can be replaced to customize the downloading process. For example, the following JavaScript\* Object Notation (JSON) represents that a file 'package.sh' will be sent to the device, and will be renamed to the value stored in the file 'package\_name', say 'linux64.sh'. Similarly, the file 'payload.bin' will be downloaded to the device and will be renamed to the value stored in the file 'payload\_name', say 'payload.bin'. Finally, the value of the file 'sh-linux64', '/bin/sh linux64.sh', will be executed as command in the device.
+## Setting up ServiceInfo transfer
+The ServiceInfo communication involves the transfer of files from the Owner to the Device.
```
Comment (trbehera) [inline docs/iot-platform-sdk/running-the-demo.md:418]: The ServiceInfo transfer involves the transfer of files from the Owner to the Device, as well as execution of some scripts in the Device.

**Review by KiranSukhavasi — DISMISSED** [review]

File: docs/iot-platform-sdk/running-the-demo.md (hunk 1 above)
Comment (TomBarnes) [reply docs/iot-platform-sdk/running-the-demo.md:467]: Git cannot render markdown syntax that is specific to mkdocs/material theme.

**Review by prateekcr — DISMISSED** [review]

**Review by TomBarnes — DISMISSED** [review]
I have reviewed and approve these changes.

**Review by prateekcr — DISMISSED** [review]

File: docs/iot-platform-sdk/running-the-demo.md [hunk 8]
```diff
@@ -454,63 +454,124 @@ The above directory contains several subdirectories, each representing a device,
...
 
-```
+These files can be a binary, an executable script or even a normal text file. The files that are to be transferred to the device is stored in the following path  `<iot-platform-sdk>/demo/ocs/config/db/v1/values`.
+
+While transferring, the files are encoded in base64 format.  The ServiceInfo configuration file  **_svi.json_** is available in the following path `<iot-platform-sdk>/demo/ocs/config/db/v1/devices/<DEVICE-GUID>`
```
Comment (KiranSukhavasi) [inline docs/iot-platform-sdk/running-the-demo.md]: extra space.

File: docs/iot-platform-sdk/running-the-demo.md [hunk 9]
```diff
@@ -454,63 +454,124 @@ The above directory contains several subdirectories, each representing a device,
...
+```
 !!! Note
-     When running the services as Docker*, only <sdo-iot-platform-sdk-root\>/demo/ocs/config/db and <sdo-iot-platform-sdk-root\>/demo/to0scheduler/config/redirect.properties, are configurable.
+     Using a single `exec` block you cannot execute multiple scripts on the Device.
+
+To avoid different configurations for different devices, a unified structure is followed for svi.json file which is described above and works for all devices.
 
-Given an OwnershipVoucher file with a unique deviceId/GUID, the following needs to be done to add it as a device in the directory <sdo-iot-platform-sdk-root\>/demo/ocs/config/db/v1/devices:
+!!! Note
+     When running the services as Docker*, only `<sdo-iot-platform-sdk-root\>/demo/ocs/config/db` and `<sdo-iot-platform-sdk-root\>/demo/to0scheduler/config/redirect.properties`, are configurable.
```
Comment (KiranSukhavasi) [inline docs/iot-platform-sdk/running-the-demo.md:522]: extra space

File: docs/iot-platform-sdk/running-the-demo.md [hunk 10]
```diff
@@ -454,63 +454,124 @@ The above directory contains several subdirectories, each representing a device,
...
 1. Create a folder named deviceId/GUID.
 
 2. Rename the Owner voucher file to 'voucher.json'. Move this file to the newly created directory.
 
 3. Create the psi.json and svi.json files under the same directory. A sample of both files are present at the sample device directory '1fae14fb-deca-405a-abdd-b25391b9d932'.
 
-The following is a sample Python\* script that, given an Owner voucher file as an argument, extracts the GUID and creates the previously mentioned device directory structure.
-
-This sample script operates under the assumption that both the script and voucher files are placed at \<sdo-iot-platform-sdk-root>/demo/ocs/config/db/v1/devices, and that this is the current working directory:
+The following is a sample Python\* script that, given an Owner voucher file as an argument, extracts the GUID and creates the previously mentioned device directory structure. This sample script operates under the assumption that both the script and voucher files are placed at `<sdo-iot-platform-sdk-root>/demo/ocs/config/db/v1/devices`, and that this is the current working directory:
```
Comment (KiranSukhavasi) [inline docs/iot-platform-sdk/running-the-demo.md]: is it Ownership Voucher in place of Owner Voucher?

**Review by KiranSukhavasi — CHANGES_REQUESTED** [review]

File: docs/iot-platform-sdk/running-the-demo.md (hunk 10 above)
Comment (DukeDavis12) [reply docs/iot-platform-sdk/running-the-demo.md]: Ownership Voucher

File: docs/iot-platform-sdk/running-the-demo.md (hunk 8 above)
Comment (DukeDavis12) [reply docs/iot-platform-sdk/running-the-demo.md]: Okay.. I will remove that

**Review by KiranSukhavasi — APPROVED** [review]

**Review by prateekcr — APPROVED** [review]

**Review by Darshini-Parikh93 — APPROVED** [review]

**Review by trbehera — APPROVED** [review]
